dali/operators/reader/webdataset_reader_op.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/webdataset_reader_op.h"
+#include <algorithm>
+#include <cstring>
+#include <string>
+
+namespace dali {
+
+void WebdatasetReader::RunImpl(HostWorkspace& ws) {
+  int num_outputs = ws.NumOutput();
+  int num_samples = GetCurrBatchSize();
+
+  for (int data_idx = 0; data_idx < num_samples; data_idx++) {
+    auto& sample = GetSample(data_idx);
+    for (int output_idx = 0; output_idx < num_outputs; output_idx++) {
+      ws.OutputRef<CPUBackend>(output_idx)[data_idx].Resize(sample[output_idx].shape());
```

Comment:
As shapes are already known how about adding `SetupImpl`?

Reply:
Is `GetSample` idempotent and/or stateless? Can we call it in advance for all samples and collect the shapes in bulk for the whole batch without sacrificing performance?

Reply:
It should, it just takes a sample from a batch that is assembled already (it is just a wrapped [] call).

Reply:
The shapes are different for each batch. Only the number of dimensions is known